### PR TITLE
td 1.2 (new formula)

### PR DESCRIPTION
**Pull request by Swatto**

### CODE REVIEW

File: Library/Formula/td.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    mkdir_p buildpath/"src/github.com/Swatto/"
+    ln_sf buildpath, buildpath/"src/github.com/Swatto/td"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "."
+    bin.install "td-#{version}" => "td"
+  end
+
+  test do
+    # Creation of the temp file
```
Comment (mikemcquaid) [inline Library/Formula/td.rb]: Just do the `JSON` require inside the test (or even the method).

File: Library/Formula/td.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    (testpath/".todos").write("[]")
```
Comment (mikemcquaid) [inline Library/Formula/td.rb]: Can skip the trailing parens.

File: Library/Formula/td.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/td.rb]: Don't need this comment.

File: Library/Formula/td.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    def read_todos
+      raw_json = (testpath/".todos").read
+      JSON.parse(raw_json)
```
Comment (mikemcquaid) [inline Library/Formula/td.rb]: Can skip the parens.

File: Library/Formula/td.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    assert_match todos[0]["desc"], "todo of test"
+    assert_match todos[0]["status"], "pending"
+
+    # Toggle a todo
+    system "#{bin}/td", "t", "1"
+    todos = read_todos
+    assert_match todos[0]["status"], "done"
+    assert_match todos.length, 1
+
+    # Clean finished todos
```
Comment (mikemcquaid) [inline Library/Formula/td.rb]: Can remove these comments and you probably only need the `Add a todo` test. Thanks!

File: Library/Formula/td.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    system "#{bin}/td", "clean"
+    todos = read_todos
+    assert_match todos.length, 0
+  end
+
+  def caveats; <<-EOS.undent
```
Comment (mikemcquaid) [inline Library/Formula/td.rb]: Can you move this method above `test do`? Thanks!

File: Library/Formula/td.rb [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    Now you need to add "TODO_DB_PATH" to your environment variable.
+    This will be use as a fallback if your directories does not contains any ".todos" files.
+
+    Find more information here:
+    #{homepage}
+    EOS
+  end
```
Comment (mikemcquaid) [inline Library/Formula/td.rb]: Just remove these caveats as they aren't Homebrew-specific, thanks!

Comment (Swatto) [inline Library/Formula/td.rb]: @mikemcquaid: What can I use to talk to the user after the installation ?

Comment (mikemcquaid) [inline Library/Formula/td.rb]: Most package managers don't. You can assume they will read the homepage.

**Comment by Swatto** [comment]
Updated with @mikemcquaid comments

**Comment by bfontaine** [comment]
JSON wasn’t available in Ruby 1.8.7, which is the default version on ML.

**Comment by Swatto** [comment]
Updated to use `utils/json` for Ruby 1.8.7

**Comment by bfontaine** [comment]
Just need a :+1: from @Homebrew/owners as this is a new formula and we’re good :smile:

**Comment by Swatto** [comment]
Ok nice ! Thanks for you help.

File: Library/Formula/td.rb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+  test do
+    require "utils/json"
+
+    (testpath/".todos").write "[]"
+
+    def read_todos
+      raw_json = (testpath/".todos").read
+      Utils::JSON.load raw_json
+    end
```
Comment (xu-cheng) [inline Library/Formula/td.rb]: Any chance don't read the json file as this should be internal format and may be changed in the future.

Instead can you just call `td search` or something?

Comment (Swatto) [inline Library/Formula/td.rb]: I've changed this to just search in the content of the file without parsing the JSON.

**Comment by Swatto** [comment]
Ok @xu-cheng, it's changed due your comment.

**Comment by xu-cheng** [comment]
Thanks for the pull request! 🎉 Homebrew depends on contributions from community members like you and we're grateful for your support.

**Pull request closed without merging**